lib/impure/rdstdin.nim

```diff
@@ -31,9 +31,27 @@ when defined(Windows):
     stdout.write(prompt)
     result = readLine(stdin, line)
 
+  proc getch(): cint {.header: "<conio.h>", importc: "_getch".}
+
+  proc readPasswordFromStdin*(prompt: string, password: var TaintedString) =
+    ## Reads a `password` from stdin without printing it. `password` must not
+    ## be ``nil``!
+    password.setLen(0)
+    var c: char
+    echo prompt
+    while true:
+       c = getch().char
+       case c
+       of '\r', chr(0xA):
+          break
+       of '\b':
+          password.setLen(result.len - 1)
+       else:
+          password.add(c)
+
 else:
-  import readline, history
-    
+  import readline, history, termios, unsigned
```

Comment:
I think the rdstdin module is pretty small as it is. Didn't want to make yet another small module with just 2 procs.